File: java/lambda/src/com/github/dannful/AniLexa/handlers/HelpIntentHandler.java
package com.github.dannful.AniLexa.handlers;

import static com.amazon.ask.request.Predicates.intentName;

import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.dispatcher.request.handler.RequestHandler;
import com.amazon.ask.model.Response;
import com.github.dannful.AniLexa.Constants;
import java.util.Optional;

public class HelpIntentHandler implements RequestHandler {

  @Override
  public boolean canHandle(HandlerInput input) {
    return input.matches(intentName("AMAZON.HelpIntent"));
  }

  @Override
  public Optional<Response> handle(HandlerInput input) {
    String speechText = "You can manage your anime list with me!";
    return input
      .getResponseBuilder()
      .withSpeech(speechText)
      .withSimpleCard(Constants.APP_ID, speechText)
      .withReprompt(speechText)
      .build();
  }
}

File: node/lambda/src/handlers/HelpIntentHandler.ts
import { HandlerInput, RequestHandler } from "ask-sdk-core";
import { Response } from "ask-sdk-model";
import { APP_ID } from "..";

export const HelpIntentHandler: RequestHandler = {
  canHandle(handlerInput: HandlerInput) {
    const request = handlerInput.requestEnvelope.request;
    return (
      request.type === "IntentRequest" &&
      request.intent.name === "AMAZON.HelpIntent"
    );
  },
  handle(handlerInput: HandlerInput): Response {
    const speakOutput = "You can manage your anime list with me";

    return handlerInput.responseBuilder
      .speak(speakOutput)
      .reprompt(speakOutput)
      .withSimpleCard(APP_ID, speakOutput)
      .getResponse();
  },
};
